pgmpy/factors/continuous/LinearGaussianCPD.py

```diff
@@ -29,106 +31,56 @@ class LinearGaussianCPD(BaseFactor):
     p(Y |x) = N(β0 + β.T * x ; σ2)
 
     """
-    def __init__(self, variable, beta, variance, evidence=[]):
+    def __init__(self, variable=None, mean=None, cov=None, evidence=[], dist=None):
         """
+        Initialize a new LinearGaussianCPD object.
+
         Parameters
         ----------
+        variable: Any hashable python object (optional)
+            The variable on which the CPD is defined.
+            Not required if dist is not None.
 
-        variable: any hashable python object
-            The variable whose CPD is defined.
+        mean: list/iterable (optional)
+            The mean of the Gaussian Distribution. For a Linear Conditional Gaussian 
+            Distribution `P(Y | A, B, C) = N(2A + 3B - 4C + 1; 1)`, the mean vector
+            should be [2, 3, -4].
+            Not required if `dist` is not None.
 
-        beta: iterable of int or float
-            An iterable representing the coefficient vector of the linear equation.
-            The first term represents the constant term in the linear equation.
+        cov: float (optional)
+            The covariance of the `varaible`.
+            Not reuired if `dist` is not None.
 
-        variance: int, float
-            The variance of the variable defined.
+        evidence: list/iterable (optional)
+            The variables on which the LinearGaussianCPD is conditioned on. 
+            Not required if `dist` is not None or there are no conditional variables.
 
-        evidence: iterable of any hashabale python objects
-            An iterable of the parents of the variable. None if there are no parents.
+        dist: Instance of pgmpy.factors.distributions.GaussianDistribution (optional)
+            The GaussianDistribution which is to be represented by the LinearGaussianCPD
+            instance. The GaussianDistribution should have only one variable.
+            Not required if all other parameters are given.
 
         Examples
         --------
+        For representing the following CPD:
```

Comment:
@ankurankan I think we should put this in comment. Allows to directly copy paste the doc in ipython console.